tests/integration/components/self-clear-cache-test.js

```diff
@@ -0,0 +1,124 @@
+import { module, skip } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { click, render, find } from '@ember/test-helpers';
+import { hbs } from 'ember-cli-htmlbars';
+
+const EPOCH_TIMESTAMP = 1626242030;
+const HUMAN_TIME = ' July 13, 2021 10:53:50 PM IST';
+const CACHE_CLEAR_CLICKED = 'CACHE_CLEAR_CLICKED';
+
+module('Integration | Component | self-clear-cache', (hooks) => {
+  setupRenderingTest(hooks);
+
+  skip('it shows last time user cleared the cache', async function (assert) {
+    assert.expect(1);
+
+    // Assemble
+    this.set('lastTime', EPOCH_TIMESTAMP);
+
+    await render(hbs`<SelfClearCache @lastClearedOn={{this.lastTime}} />`);
+
+    // Assert
+    assert.dom('[data-test-last-time]').hasText(HUMAN_TIME);
+  });
+
+  skip('it triggers cache clear on button click', async function (assert) {
+    assert.expect(1);
+
+    // Assemble
+    this.set('lastTime', EPOCH_TIMESTAMP);
+
+    this.set('onCacheClear', function () {
+      assert.step(CACHE_CLEAR_CLICKED);
+    });
+
+    await render(hbs`
+      <SelfClearCache
+        @lastClearedOn={{this.lastTime}}
+        @onCacheClear={{this.onCacheClear}}
+      />
+    `);
+
+    // Act
+    const btn = find('[data-test-btn-clear-cache]');
+    await click(btn);
+
+    // Assert
+    assert.verifySteps([CACHE_CLEAR_CLICKED]);
+  });
+
+  skip('it disables button after a click', async function (assert) {
```

Comment:
On what condition it gets enabled again?

Reply:
I had the same doubt, need to ask @ankushdharkar, but probably there will be a fixed time after which the button gets enabled again and it will be handled on the frontend, we need not to test that I guess

Reply:
How I think it should be -
Button clicked > Handle Click function calls which disables button & an API request to clear cache > Wait for server to respond > Server responds > Enable button again

Reply:
Good questions and suggestions.

We should be testing this on **Frontend** **AND** **Backend**. Backend should always validate and verify the behaviour (otherwise a _bad_ user can just use curl/postman/etc to trigger the API anyway)

The button to be disabled here is to prevent multiple calls from firing (because clearing cache costs us as well as, it will cost crypto to clear your cache in future). We want to ideally make idempotent calls to clear the cache, and we should discuss this. We would have to implement some sort of UUID for this.

You can read more about the solution here: https://stripe.com/docs/api/idempotent_requests, but until then, we should just do a simpler implementation of maybe disabling the button for 5 seconds and move on.

- [ ] Start an RFC for the the idempotent requests setup

Reply:
Created an RFC here
https://github.com/Real-Dev-Squad/website-backend/issues/627